src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/RoutePatternCompletionProvider.cs

```diff
@@ -252,19 +256,24 @@ private static void ProvidePolicyNameCompletions(EmbeddedCompletionContext conte
         context.AddIfMissing("float", suffix: null, "Matches a valid float value in the invariant culture.", WellKnownTags.Keyword, parentOpt);
         context.AddIfMissing("guid", suffix: null, "Matches a valid Guid value.", WellKnownTags.Keyword, parentOpt);
         context.AddIfMissing("long", suffix: null, "Matches any 64-bit integer.", WellKnownTags.Keyword, parentOpt);
-        context.AddIfMissing("minlength", suffix: null, "Matches a string with a length greater than, or equal to, the constraint argument.", WellKnownTags.Keyword, parentOpt);
-        context.AddIfMissing("maxlength", suffix: null, "Matches a string with a length less than, or equal to, the constraint argument.", WellKnownTags.Keyword, parentOpt);
-        context.AddIfMissing("length", suffix: null, @"The string length constraint supports one or two constraint arguments.
+
+        // The following constraints are only available for HTTP route matching.
+        if (context.Options == RouteOptions.Http)
```

Comment:
This change is part of supporting component routes. I made it in this PR so I'd have a way to test the option is being detected.